homeassistant/components/sensibo/climate.py

```diff
@@ -176,9 +176,7 @@ async def async_set_temperature(self, **kwargs) -> None:
             raise HomeAssistantError(
                 "Current mode doesn't support setting Target Temperature"
             )
-
-        if (temperature := kwargs.get(ATTR_TEMPERATURE)) is None:
-            return
+        temperature = kwargs.get(ATTR_TEMPERATURE)
```

Comment:
Technically the temperature parameter is optional.

https://github.com/home-assistant/core/blob/4628b151ffa49387e2ff8e7b5dafb6e7cf7d85f5/homeassistant/components/climate/__init__.py#L95-L107

Reply:
Just wrote it https://github.com/home-assistant/core/pull/71149#discussion_r862488938 
I think we should consider raising `HomeAssistantError` if temperature is missing since there is no use case for this service without temperature.

Reply:
We normally raise ValueError on bad user input to service calls.

Reply:
As written above I didn't find a case where it could actually be None but now I have one so will add it into the tests instead. https://github.com/home-assistant/core/pull/71149#discussion_r862489800

Reply:
btw, looking at the code base most integrations just return here, I think it serves as a bad example and probably copy pasted between integrations. IMO user should be notified if a service call does not perform the indented operation due to missing parameters.